components/button/floating-buttons.js

```diff
@@ -0,0 +1,207 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+import { styleMap } from 'lit-html/directives/style-map.js';
+
+class FloatingButtons extends RtlMixin(LitElement) {
+
+	static get properties() {
+		return {
+			alwaysFloat: { type: Boolean, attribute: 'always-float' },
+			minHeight: { type: String, attribute: 'min-height' },
+			_containerMarginLeft: { type: Object },
+			_containerMarginRight: { type: Object },
+			_floating: { type: Boolean, reflect: true },
+			_innerContainerLeft: { type: Number },
+			_innerContainerRight: { type: Number }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				box-sizing: border-box;
+				display: block;
+			}
+
+			:host([_floating]) {
+				bottom: -10px;
+				left: 0;
+				position: -webkit-sticky;
+				position: sticky;
+				right: 0;
+				z-index: 999;
+			}
+
+			:host([hidden]) {
+				display: none;
+			}
+
+			.d2l-floating-buttons-container {
+				border-top: 1px solid transparent;
+				display: block;
+				overflow-x: hidden;
+			}
+
+			:host([_floating]) .d2l-floating-buttons-container {
+				background-color: #ffffff;
+				background-color: rgba(255, 255, 255, 0.88);
+				border-top-color: var(--d2l-color-mica);
+				box-shadow: 0 -2px 4px rgba(86, 90, 92, .2);
+				transform: translate(0, -10px);
+				transition: transform 500ms, border-top-color 500ms, background-color 500ms;
+			}
+
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+
+			.d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			.d2l-floating-buttons-inner-container ::slotted(button),
+			.d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-right: 0.75rem !important;
+				margin-bottom: 0.75rem !important;
+			}
+
+			:host([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			:host([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(button),
+			:host([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-left: 0.75rem !important;
+				margin-right: 0 !important;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.minHeight = '500px';
+		this._containerTop = null;
+
+		this._calcIfFloating = this._calcIfFloating.bind(this);
+		this._calcContainerPosition = this._calcContainerPosition.bind(this);
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		window.addEventListener('scroll', this._calcContainerPosition);
+		window.addEventListener('resize', this._calcContainerPosition);
+		window.addEventListener('d2l-tab-panel-selected', this._calcContainerPosition);
```

Comment:
This is for dealing with tabs based upon how the tabs web component works. Not sure if there is a better option.